packages/flame/lib/src/widgets/animation_widget.dart

```diff
@@ -17,7 +18,7 @@ class SpriteAnimationWidget extends StatelessWidget {
   /// Should the animation be playing or not
   final bool playing;
 
-  final Future<SpriteAnimation> Function() _animationFuture;
+  final FutureOr<SpriteAnimation> Function() _animationFuture;
```

Comment:
This looks kinda suspicious to me -- this variable claims to be a `Future`, but really it's a function generating futures. So, then the question is: how often will this function be called, how many futures will be produced?

I think it would be much clearer if this was an actual `FutureOr` here, instead of a function.

Reply:
I totally agree with this. The function is called only once.

But I give up with
> It should be something that I don't know. 

---
I will try this right away.